pkg/users/account-create-dialog.js

```diff
@@ -162,31 +208,59 @@
     return remove_diacritics(result);
 }
 
-export function account_create_dialog(accounts) {
+export function account_create_dialog(accounts, shells) {
     let dlg = null;
     const state = {
+        dialogLoading: true,
         real_name: "",
         user_name: "",
         password: "",
         password_confirm: "",
         locked: false,
         confirm_weak: false,
         change_passw_force: false,
+        base_home_dir: "",
+        home_dir: "",
+        home_dir_dirty: false,
+        shell: "",
+        isShellSelectExpanded: false,
     };
     let errors = { };
 
     let old_password = null;
     let user_name_dirty = false;
 
+    function get_defaults() {
+        return cockpit.spawn(["/usr/sbin/useradd", "-D"], { superuser: "require" })
+                .catch(() => "")
+                .then(defaults => {
+                    defaults.split("\n").forEach(item => {
+                        if (item.indexOf("SHELL=") === 0) {
+                            change("shell", item.split("=")[1] || "/bin/bash");
+                        } else if (item.indexOf("HOME=") === 0) {
+                            change("base_home_dir", item.split("=")[1] || "");
+                        }
+                    });
+                })
+                .finally(() => change("dialogLoading", false));
+    }
+
     function change(field, value) {
         state[field] = value;
         errors = { };
 
-        if (field == "user_name")
+        if (field == "user_name") {
             user_name_dirty = true;
+            if (!state.home_dir_dirty)
+                state.home_dir = get_default_home_dir(state.base_home_dir, value);
+        }
```

Comment:
These 3 added lines are not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-18234-20230202-125519-871b841d-fedora-37-devel//Coverage/lcov/github-pr.diff.gcov.html)